gwcs/coordinate_frames.py

```diff
@@ -512,14 +520,13 @@ def coordinates(self, *args):
         return self._convert_to_time(dt, unit=self.unit[0], **self._attrs)
 
     def _convert_to_time(self, dt, *, unit, **kwargs):
-        if not isinstance(self.reference_frame.value, np.ndarray):
-            if not hasattr(dt, 'unit'):
-                dt = dt * unit
-            return self.reference_frame + dt
-
-        else:
+        if isinstance(dt, time.Time) or isinstance(self.reference_frame.value, np.ndarray):
```

Comment:
Could `dt` be a `TimeDelta` object?
It seems there's no other restriction for this except the check above.